types/rdfjs__data-model/lib/fromTerm.d.ts

```diff
@@ -0,0 +1,14 @@
+import * as RDF from '@rdfjs/types';
+
+export interface FromTerm<T extends RDF.Term, D extends RDF.DataFactory> {
+    (factory: D, original: T):
+        T extends RDF.Literal ? ReturnType<D['literal']>
+            : T extends RDF.BlankNode ? ReturnType<D['blankNode']>
+                : T extends RDF.NamedNode ? ReturnType<D['namedNode']>
+                    : T extends RDF.Variable ? D['variable'] extends (...arg: any[]) => RDF.Variable ? ReturnType<D['variable']> : never
+                        : T extends RDF.Quad ? ReturnType<D['quad']>
+                            : T extends RDF.DefaultGraph ? ReturnType<D['defaultGraph']>
+                                : never;
```

Comment:
It looks like this will infer `never` if I input something of type `BaseQuad` that does not satisfy the `Quad` type. Persumably it should be `T extends RDF.BaseQuad` instead?

Reply:
Good observation. Unfortunately, now that tried to improve the tests, it appears that it always returns `BaseQuad`... 🙄

Reply:
Ok, sorry, some GitHub fluke did not allow me to edit my reply.  The factory itself has `BaseQuad` as the generic arg so guess it's correct.